python/src/aiconfig/editor/client/src/components/prompt/RunPromptButton.tsx

```diff
@@ -15,39 +16,49 @@ export default memo(function RunPromptButton({
   isRunning = false,
   disabled = false,
 }: Props) {
-  const onClick = async () => {
+  const { readOnly } = useContext(AIConfigContext);
+  const disabledOrReadOnly = disabled || readOnly;
+
+  const onClick = useCallback(async () => {
     if (isRunning) {
       return await cancel();
     } else {
       return await runPrompt();
     }
-  };
-  const button = (
-    <Button
-      onClick={onClick}
-      disabled={disabled}
-      p="xs"
-      size="xs"
-      className="runPromptButton"
-    >
-      {isRunning ? (
-        <Flex align="center" justify="center">
-          <Loader style={{ position: "absolute" }} size="xs" color="white" />
-          <IconPlayerStop fill="white" size={14} />
-        </Flex>
-      ) : (
-        <>
-          <IconPlayerPlayFilled size="16" />
-        </>
-      )}
-    </Button>
-  );
+  }, [cancel, runPrompt, isRunning]);
+
+  const button = useMemo(() => {
+    return (
+      <Button
+        onClick={onClick}
+        disabled={disabledOrReadOnly}
+        p="xs"
+        size="xs"
+        className="runPromptButton"
+      >
+        {isRunning ? (
+          <Flex align="center" justify="center">
+            <Loader style={{ position: "absolute" }} size="xs" color="white" />
+            <IconPlayerStop fill="white" size={14} />
+          </Flex>
+        ) : (
+          <>
+            <IconPlayerPlayFilled size="16" />
+          </>
+        )}
+      </Button>
+    );
+  }, [disabledOrReadOnly, onClick, isRunning]);
+
+  const disabledButton = useMemo(() => {
```

Comment:
Let's not memoize these or use `useCallback` since that is actually going to result in extra unnecessary processing. This `RunPromptButton` is already memoized, so it will not re-render unless the props change. Then if that happens, note that the deps chain will result in button's useMemo always creating a new button anyway. So, this just results in extra processing to compute/compare the dependency changes within the RunPromptButton every time it re-renders